### ISSUE TITLE
distance_to_patches: block count scales with cells × wall faces, not just cells — minutes per call at ~1M cells

### ISSUE THREAD

**Issue by ctdegroot** [DeGrootResearchGroup/aquaflux#240]
`aquaflux.mesh.distance.distance_to_patches` — the near-wall closure's nearest-face search, used by
`SSTTurbulence.build` — blocks its search to bound peak memory, but the block size shrinks as the
number of *target* (wall) faces grows, so on a case with several wall patches at a large mesh size the
block count explodes into tens of thousands, each a separate eagerly-dispatched op, followed by one
`jnp.concatenate` over all of them. At scales relevant to the meshes this project is now planning
(~1M cells), that's tens of seconds to minutes for one call.

## Impact

Case assembly (`SSTTurbulence.build`, called once per Reynolds-continuation rung — see
`compare.py`'s docstring: "each rung rebuilds both the case ... and its continuation") pays this cost
repeatedly on any case with more than a couple of wall patches, and it gets worse as mesh size grows.
It does not show up on the current validation cases (`bfs3d` is 23,040 cells / 4,736 wall faces — few
enough blocks to be invisible) — this is a scaling problem, not a correctness one, which is presumably
why it hasn't been caught before.

## Reproduce

```python
import jax.numpy as jnp
from aquaflux.mesh.distance import distance_to_patches, _nearest_squared_distance, _SEARCH_WORKING_BYTES
from aquaflux.mesh.structured import structured_grid_3d

mesh = structured_grid_3d(100, 100, 100, named_boundaries=True)  # 1,000,000 cells
geom = mesh.geometry()
patches = ["top", "bottom", "front", "back"]  # 4 wall patches, 40,000 combined wall faces

n_wall = sum(mesh.face_patches.indices(p).shape[0] for p in patches)
per_cell = 3 * n_wall * 3 * 8
block = max(1, min(mesh.n_cells, _SEARCH_WORKING_BYTES // per_cell))
print(block, -(-mesh.n_cells // block))  # 23  43479

d = distance_to_patches(mesh, geom, patches)  # ~77s wall clock
```

## Evidence

Measured on a synthetic `structured_grid_3d(100, 100, 100, named_boundaries=True)` (1,000,000 cells)
with 4 named wall patches (40,000 combined wall faces) — a wall/cell ratio comparable to `bfs3d`'s own
(`upperWall`/`lowerWall`/`sideWalls`), just at a bigger mesh:

| | |
|---|---|
| block size (`_SEARCH_WORKING_BYTES // per_cell`) | 23 cells |
| number of blocks | 43,479 |
| wall time, one `distance_to_patches` call | **77 s** |

A same-scale run of the identical call sequence inside a full case build separately triggered JAX's
own slow-compile alarm on the block loop's `jnp.concatenate`:

```
[Compiling module jit_concatenate for CPU] Very slow compile?
The operation took 5m5.656426s
```

Configuration: aquaflux `e1a34d1`, Python 3.13.4, JAX 0.10.2, macOS/arm64 CPU backend, x64 enabled. Not
yet measured on the real `read_openfoam` mesh-import path (only on a synthetic structured mesh standing
in for it) or on a GPU backend.

## Root cause

`_nearest_squared_distance` (`aquaflux/mesh/distance.py`) sizes each block from a **fixed byte budget**
divided by the **per-cell cost at the current target-face count**:

```python
per_cell = max(1, 3 * n_target * dim * centroid.dtype.itemsize)
block = max(1, min(n_cells, _SEARCH_WORKING_BYTES // per_cell))
```

That correctly bounds peak memory per block — which is what it was built for (see the module
docstring: the un-blocked whole-array form needs "a little over 6 GB live at once" on `bfs3d`'s own
23,040 cells / 4,736 wall faces). But the number of blocks is then
`n_cells * n_target / (_SEARCH_WORKING_BYTES / (3 * dim * itemsize))` — it grows with the *product* of
cell count and wall-face count, not cell count alone. The mechanism was designed to bound memory, not
block count, and block count is what drives wall-clock cost here: each block is a separate eager
dispatch, and the final `jnp.concatenate` over all of them is a single op whose compile cost appears to
scale with the number of operands rather than the total data size.

## Possible directions (not investigated in depth)

- Batch the blocked loop through `jax.lax.map` / a fixed-shape `jax.lax.scan` writing into a
  preallocated output array instead of a Python list + one large `jnp.concatenate` — one compiled
  program executed N times, no giant-arity concatenate.
- A proper spatial index (k-d tree / grid binning) for the *concrete*-coordinate case, falling back to
  today's blocked brute-force only when the geometry is traced — the module docstring notes this is
  needed because `node_coords` can be a differentiable leaf during a gradient-of-mesh build, which rules
  out a plain `scipy.spatial.cKDTree` as the only path.
